### PR TITLE
Allow service principals with double-quote (") and backslash (\)

### LINKED ISSUE #3038: deal with backtick (`) chars in sp keys
Exact details TBD, but from a high level we should either fail validation or escape backticks. Observation suggests that currently we're not dealing with it at all which yields non-deterministic side-effects during deployment.

### PR DESCRIPTION
**Pull request by tariq1890**
This fixes an issue when passwords contain JSON special chars like " and \

Fixes #3038

### CODE REVIEW

File: parts/k8s/kubernetesconfigs.sh [hunk 1]
```diff
@@ -131,6 +131,10 @@ function configureK8s() {
     chmod 0600 "${AZURE_JSON_PATH}"
     chown root:root "${AZURE_JSON_PATH}"
 
+    # Perform the required JSON escaping for special characters " an \
```
Comment (jackfrancis) [inline parts/k8s/kubernetesconfigs.sh]: s/an/and

Comment (tariq1890) [reply parts/k8s/kubernetesconfigs.sh]: lol

File: parts/k8s/kubernetesconfigs.sh [hunk 2]
```diff
@@ -131,6 +131,10 @@ function configureK8s() {
     chmod 0600 "${AZURE_JSON_PATH}"
     chown root:root "${AZURE_JSON_PATH}"
 
+    # Perform the required JSON escaping for special characters " and \
```
Comment (CecileRobertMichon) [inline parts/k8s/kubernetesconfigs.sh]: let's move this below `set +x` on line 138

**Comment by tariq1890** [comment]
@CecileRobertMichon Fixed :)

**Comment by CecileRobertMichon** [comment]
/lgtm

oops

**Pull request merged**